pkg/nodeconfig/payload/payload.go

```diff
@@ -175,7 +175,34 @@ $cni_template=$cni_template.Replace("provider_address",$provider_address)
 Compare-And-Replace-Config -ConfigPath CNI_CONFIG_PATH -NewConfigContent $cni_template
 
 # Create HNS endpoint if it doesn't exist
-$endpoint = Invoke-HNSRequest GET endpoints | where { $_.Name -eq 'VIPEndpoint'}
+
+$retryCount = 20
+$retryDelay = 10
+$attempt = 0
+
+while ($attempt -lt $retryCount) {
+  try {
+    $hns_network=Get-HnsNetwork  | where { $_.Name -eq 'HNS_NETWORK'}
+    $endpoint = Invoke-HNSRequest GET endpoints | where { $_.Name -eq 'VIPEndpoint'}
+
+    # Check if endpoint is null
+    if ($null -eq $endpoint) {
```

Comment:
Im confused as to why this is being done, especially given the code below,
```
if( $endpoint -eq $null) {
    $endpoint = New-HnsEndpoint -NetworkId $hns_network.ID -Name "VIPEndpoint"
    Attach-HNSHostEndpoint -EndpointID $endpoint.ID -CompartmentID 1
}
```
And the comment above `# Create HNS endpoint if it doesn't exist`

Reply:
Shouldn't you be checking if hns_network is null?

Reply:
Please make this a function after you have corrected the behavior.

Reply:
Specifically the function would be to encapsulate the retry logic, returning the object we care about so the script is more easily readable.